src/ray/gcs/redis_context.cc

```diff
@@ -345,10 +346,22 @@ Status ConnectWithRetries(const std::string &address, int port,
   return Status::OK();
 }
 
+template <typename RedisContext, typename RedisConnectFunction>
+Status ConnectOrFail(const std::string &address, int port,
+                     const RedisConnectFunction &connect_function,
+                     RedisContext **context) {
+  std::string errorMessage = "";
+  Status status = ConnectWithRetries(address, port, connect_function, context);
+  if (!status.ok()) {
```

Comment:
Tiny: check this status with `Status::RedisError`

Reply:
> check this status with `Status::RedisError`

Sorry, what? I don't understand.

...Do you mean `IsRedisError()`, perhaps? If that's what you mean, shouldn't it fail no matter what kind of error? (We might later want to return different kinds of errors.)

Reply:
Yeah. For `RedisError` we should fail but for other tpyes of error we might do other actions(now we only have one error so that's an assumption).
For now it looks like we should fail whatever error happens, but i think it's a good practice to tell difference between types of error, for future changes.